pkg/drivers/generic/generic.go

```diff
@@ -360,3 +364,14 @@ func (d *Generic) Insert(ctx context.Context, key string, create, delete bool, c
 	err = row.Scan(&id)
 	return id, err
 }
+
+func (d Generic) applyLimit(sql string, limit int64) string {
```

Comment:
Would it be cleaner to move this into the driver implementations rather than having generic introspect the name and change behavior? Feels a bit like a leaky abstraction to do this here.

Reply:
Sure.. will rework this

Reply:
@brandond  i have made some changes. Please let me know what you think..